provider/identity_provider_resource.go

```diff
@@ -103,28 +110,138 @@ func (t *IdentityProviderResourceType) gitlabSchema() tfsdk.NestedAttributes {
+			Type: types.ListType{
+				ElemType: types.StringType,
+			},
+			Optional: true,
+		},
+		"ca": {
+			Type:        types.StringType,
+			Description: "Path to PEM-encoded certificate file to use when making requests to the server.",
+			Optional:    true,
+		},
+		"client_id": {
+			Description: "Client identifier of a registered Github OAuth application.",
+			Type:        types.StringType,
+			Required:    true,
+		},
+		"client_secret": {
+			Description: "Client secret issued by Github.",
+			Type:        types.StringType,
+			Required:    true,
+			Sensitive:   true,
 		},
 	})
 }
 
+func (t *IdentityProviderResourceType) githubValidators() []tfsdk.AttributeValidator {
+	errSumm := "Invalid GitHub IDP resource configuration"
+	return []tfsdk.AttributeValidator{
+		&attributeValidator{
+			description:   "GitHub IDP requires either organizations or teams",
+			mdDescription: "",
+			validate: func(ctx context.Context, req tfsdk.ValidateAttributeRequest, resp *tfsdk.ValidateAttributeResponse) {
+				ghState := &GithubIdentityProvider{}
+				diag := req.Config.GetAttribute(ctx, req.AttributePath, ghState)
+				if diag.HasError() {
+					// No attribute to validate
+					return
+				}
+				// At only one restriction plan is required
+				if len(ghState.Teams) == 0 && len(ghState.Organizations) == 0 {
+					resp.Diagnostics.AddError(errSumm, "GitHub IDP requires missing attributes 'organizations' OR 'teams'")
+				}
+				if len(ghState.Teams) > 0 && len(ghState.Organizations) > 0 {
+					resp.Diagnostics.AddError(errSumm, "GitHub IDP requires either 'organizations' or 'teams', not both.")
+				}
+			},
+		},
+		&attributeValidator{
+			description:   "GitHub IDP teams format validator",
+			mdDescription: "",
+			validate: func(ctx context.Context, req tfsdk.ValidateAttributeRequest, resp *tfsdk.ValidateAttributeResponse) {
+				ghState := &GithubIdentityProvider{}
+				diag := req.Config.GetAttribute(ctx, req.AttributePath, ghState)
+				if diag.HasError() {
+					// No attribute to validate
+					return
+				}
+				// Validate teams format
+				for i, team := range ghState.Teams {
+					parts := strings.Split(team, "/")
+					if len(parts) != 2 {
+						resp.Diagnostics.AddError(errSumm,
+							fmt.Sprintf("Expected a GitHub team to follow the form '<org>/<team>', Got %v at index %d",
+								team, i),
+						)
+						return
+					}
+				}
+			},
+		},
+		&attributeValidator{
+			description:   "Github IDP hostname validator",
+			mdDescription: "",
+			validate: func(ctx context.Context, req tfsdk.ValidateAttributeRequest, resp *tfsdk.ValidateAttributeResponse) {
+				ghState := &GithubIdentityProvider{}
+				diag := req.Config.GetAttribute(ctx, req.AttributePath, ghState)
+				if diag.HasError() {
+					// No attribute to validate
+					return
+				}
+				// Validate hostname
+				if !ghState.Hostname.Unknown && !ghState.Hostname.Null && len(ghState.Hostname.Value) > 0 {
```

Comment:
ghState.Hostname is from Type:        types.StringType, I think the validation is incorrect

Reply:
Will have the same result, an empty string, is of length 0. I can change it if you prefer this way of checking.